tests/test_cli_textconv.py

```diff
@@ -0,0 +1,288 @@
+``sphobjinv`` expects an INFILE inventory, converts it, then writes to
+stdout. Intended for use with git diff. git, detect changes, by first
+converting an (partially binary) inventory to plain text.
+
+**Author**
+    Dave Faulkmore (msftcangoblowme@protonmail.com)
+
+**File Created**
+    23 Aug 2024
+
+**Copyright**
+    \(c) Brian Skinn 2016-2024
+
+**Source Repository**
+    http://www.github.com/bskinn/sphobjinv
+
+**Documentation**
+    https://sphobjinv.readthedocs.io/en/stable
+
+**License**
+    Code: `MIT License`_
+
+    Docs & Docstrings: |CC BY 4.0|_
+
+    See |license_txt|_ for full license terms.
+
+.. code-block:: shell
+
+   pytest --showlocals --cov=sphobjinv --cov-report=term-missing \
+   --cov-config=pyproject.toml --nonloc tests
+
+**Members**
+
+"""
+
+
+import json
+import os
+import shlex
+import subprocess as sp  # noqa: S404
+
+import pytest
+from stdio_mgr import stdio_mgr
+
+from sphobjinv import Inventory
+from sphobjinv import SourceTypes
+from sphobjinv.fileops import readbytes
+
+CLI_TEST_TIMEOUT = 2
+# Is an entrypoint, but not a package
+CLI_CMDS = ["sphobjinv-textconv"]
+
+pytestmark = [pytest.mark.cli, pytest.mark.local]
+
+
+class TestTextconvMisc:
+    """Tests for miscellaneous CLI functions."""
+
+    @pytest.mark.timeout(CLI_TEST_TIMEOUT)
+    @pytest.mark.parametrize("cmd", CLI_CMDS)
+    def test_cli_textconv_help(self, cmd, run_cmdline_no_checks):
+        """Confirm that actual shell invocations do not error.
+
+        .. code-block:: shell
+
+           pytest --showlocals --cov=sphobjinv --cov-report=term-missing \
+           --cov-config=pyproject.toml -k test_cli_textconv_help tests
+
+        """
+        runargs = shlex.split(cmd)
+        runargs.append("--help")
+
+        with stdio_mgr() as (in_, out_, err_):
+            retcode, is_sys_exit = run_cmdline_no_checks(runargs)
+            str_out = out_.getvalue()
+            assert "sphobjinv-textconv" in str_out
+
+            # Ideally, the only place sys.exit calls occur within a codebase is in
+            # entrypoint file(s). In this case, sphobjinv.cli.core
+            #
+            # Each unique custom Exception has a corresponding unique exit code.
+            #
+            # Testing looks at exit codes only.
+            #
+            # Not the error messages, which could change or be localized
+            #
+            # In command line utilities, relaying possible errors is common practice
+            #
+            # From an UX POV, running echo $? and getting 1 on error is
+            # useless and frustrating.
+            #
+            # Not relaying errors and giving exact feedback on how to rectify
+            # the issue is bad UX.
+            #
+            # So if the numerous exit codes of 1 looks strange. It is; but this is
+            # a separate issue best solved within a dedicated commit
+            assert f"EXIT CODES{os.linesep}" in str_out
```

Comment:
I'll definitely be taking a closer look at this, for the project as a whole. I appreciate you pointing it out.

Could you actually duplicate these thoughts into a new ticket on the project?

(I'm aware that the exception handling in the project is substandard and intend to do something about it eventually (see #118), but it hadn't occurred to me to tie an exception hierarchy directly to the CLI exit codes in this way.)

Reply:
Thanks for accepting the exception handling in cli entrypoints as a worthy issue.

Will create a new issue(s) ... after completing #295